src/browser/main.coffee

```diff
@@ -69,7 +69,9 @@ setupCrashReporter = ->
 setupAtomHome = ->
   return if process.env.ATOM_HOME
 
-  atomHome = path.join(app.getHomeDir(), '.atom')
+  atomHome = path.join(path.dirname(process.execPath), 'userdata')
```

Comment:
`process.execPath` is something like `/Applications/Atom.app/Contents/Frameworks/Atom Helper.app/Contents/MacOS/Atom Helper` on Mac. This folder will be completely replaced whenever Atom automatically updates which means the `userdata` folder, if there, would be deleted.

This needs to be supported in a way where the config folder is not in a location that is deleted by automatic updates.